strawberry_django/filters.py

```diff
@@ -119,6 +121,18 @@ def build_filter_kwargs(filters):
     return filter_kwargs, filter_methods
 
 
+@functools.cache()
```

Comment:
I have not finished investigating, but am seeing instances running out of memory due to upgrading strawberry-graphql-django from 0.5.4 to 0.7, with this change being added in 0.6..

Since this is basically an unbound `lru_cache`, I suggest adding some `max_size` here at least - but have no idea currently why it was added in the first place etc..

Will try with 0.6 to harden / narrow down the changes that might cause the leaking / increasing memory usage.

Reply:
Do you have too much filter methods in your schema?

I have nothing against adding a `max_size` here. Maybe a `max_size` of ~200 is enough?

Reply:
There shouldn't be too many, but it looks more like e.g. lambdas being passed in there or something similar.
Maybe due to https://github.com/blb-ventures/strawberry-django-plus?

Can investigate more next week.. (it is also an issue with 0.6, making this change more likely to cause it)

Reply:
Just investigated here, it was due to bound methods being different every time (for obvious reasons). Just fixed it here and added a max_size to avoid any issue alike in the future: https://github.com/strawberry-graphql/strawberry-graphql-django/commit/dd09948f35849ae442ab75fc4fa5c171b8cdfbb1